vms/subnet.go

```diff
@@ -0,0 +1,257 @@
+	vm CustomVM,
+	network network.Network,
+	privateKey string,
+) error {
+	log.Info("creating subnet")
+
+	userPass := api.UserPass{
+		Username: "test",
+		Password: "vmsrkewl",
+	}
+
+	allNodes, err := network.GetAllNodes()
+	if err != nil {
+		return err
+	}
+
+	txNodeNames, err := network.GetNodeNames()
+	if err != nil {
+		return err
+	}
+	// txNode will be the node we issue all transactions on
+	txNode := allNodes[txNodeNames[0]]
+	txClient := txNode.GetAPIClient()
+	// we need to create a user for platformvm calls
+	ok, err := txClient.KeystoreAPI().CreateUser(userPass)
+	if !ok || err != nil {
+		return fmt.Errorf("could not create user: %w", err)
+	}
+
+	txPChainClient := txClient.PChainAPI()
+	// Import genesis key
+	fundedAddress, err := txPChainClient.ImportKey(userPass, privateKey)
+	if err != nil {
+		return fmt.Errorf("unable to import genesis key: %w", err)
+	}
+	balance, err := txPChainClient.GetBalance(fundedAddress)
+	if err != nil {
+		return fmt.Errorf("unable to get genesis key balance: %w", err)
+	}
+	log.Info("found %d on address %s", balance.Balance, fundedAddress)
+
+	// Create a subnet
+	subnetIDTx, err := txPChainClient.CreateSubnet(
+		userPass,
+		[]string{fundedAddress},
+		fundedAddress,
+		[]string{fundedAddress},
+		defaultKeyThreshold,
+	)
+	if err != nil {
+		return fmt.Errorf("unable to create subnet: %w", err)
+	}
+
+CREATE_SUBNET:
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-time.After(loopTimeout):
+			status, _ := txPChainClient.GetTxStatus(subnetIDTx, true)
+			if status.Status == platformvm.Committed {
+				break CREATE_SUBNET
+			}
+			log.Debug("waiting for subnet creation tx (%s) to be accepted", subnetIDTx)
+		}
+	}
+	log.Info("subnet creation tx (%s) accepted", subnetIDTx)
+
+	// Confirm created subnet appears in subnet list
+	subnets, err := txPChainClient.GetSubnets([]ids.ID{})
+	if err != nil {
+		return fmt.Errorf("cannot query subnets: %w", err)
+	}
+
+	var rSubnetID ids.ID
+	var subnetID string
+	for _, s := range subnets {
+		if s.ID.String() == vm.SubnetID {
+			rSubnetID = s.ID
+			subnetID = s.ID.String()
+			break
+		}
+	}
+	if subnetID != vm.SubnetID {
+		return fmt.Errorf("expected subnet %s but got %s", vm.SubnetID, subnetID)
+	}
+
+	// Add all validators to subnet with equal weight
+	for _, n := range allNodes {
+		nodeID := n.GetNodeID().PrefixedString(constants.NodeIDPrefix)
+		txID, err := txPChainClient.AddSubnetValidator(
+			userPass, []string{fundedAddress}, fundedAddress,
+			subnetID, nodeID, validatorWeight,
+			uint64(time.Now().Add(validatorStartDiff).Unix()),
+			uint64(time.Now().Add(validatorEndDiff).Unix()),
+		)
+		if err != nil {
+			return fmt.Errorf("unable to add subnet validator: %w", err)
+		}
+
+	ADD_VALIDATOR:
+		for {
+			select {
+			case <-ctx.Done():
+				return ctx.Err()
+			case <-time.After(loopTimeout):
+				status, _ := txPChainClient.GetTxStatus(txID, true)
+				if status.Status == platformvm.Committed {
```

Comment:
Again, can get status outside of select and avoid label. Also, it should have a helper in this file for this kind of tx wait.